src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -193,6 +194,65 @@ def set_key(cmd,
         "Fail to set the key '{}' due to a conflicting operation.".format(key))
 
 
+def set_keyvault(cmd,
+                 key,
+                 secret_identifier,
+                 name=None,
+                 label=None,
+                 tags=None,
+                 yes=False,
+                 connection_string=None):
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+
+    keyvault_ref_value = json.dumps({"uri": secret_identifier}, ensure_ascii=False)
+    retry_times = 3
```

Comment:
why we need json dumps here to create a dictionary?

Reply:
'value' attribute of KeyValue object is supposed to be a string. In order to preserve double quotes while converting this dict to str, we use json dumps.

---
In reply to: [358477565](https://github.com/Azure/azure-cli/pull/11571#discussion_r358477565) [](ancestors = 358477565)

Reply:
Okay, so the key value will look like
{
"Key":"K1"
"Value":"{"uri":"identitfier"}"
} ?

---
In reply to: [358518179](https://github.com/Azure/azure-cli/pull/11571#discussion_r358518179) [](ancestors = 358518179,358477565)

Reply:
Actually, like this:

{
"key": "K1",
"value": "{\"uri\": \"[https://contoso.vault.azure.net/Secrets/DummySecret/Dummyversion\](https://contoso.vault.azure.net/Secrets/DummySecret/Dummyversion\)"}"
}

---
In reply to: [358528290](https://github.com/Azure/azure-cli/pull/11571#discussion_r358528290) [](ancestors = 358528290,358518179,358477565)